rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -2073,11 +2073,69 @@ public int registerBtcTransfer(
             Address userRefundAddress,
             RskAddress lbcAddress,
             Address lpbtcAddress,
-            boolean executionStatus
-    ) {
+            boolean shouldTransferToContract
+    ) throws BlockStoreException, RegisterBtcTransferException, IOException {
+        if (!BridgeUtils.isContractTx(rskTx)) {
+            String errorMesage = String.format("[registerBtcTransfer] [rskTx:%s] Transaction not a contract", ByteUtil.toHexString(rskTx.getHash().getBytes()));
+            logger.warn(errorMesage);
+            throw new RegisterBtcTransferContractValidationException(errorMesage);
+        }
+
+        BtcTransaction btcTx = new BtcTransaction(bridgeConstants.getBtcParams(), btcTxSerialized);
+        Coin totalAmount = this.getAmountToActiveFederation(btcTx);
+
+        Sha256Hash btcTxHash = BtcTransactionFormatUtils.calculateBtcTxHash(btcTxSerialized);
+
+        if (!validationsForRegisterBtcTransaction(btcTxHash, height, pmtSerialized, btcTxSerialized)) {
+            String errorMesage = String.format("[registerBtcTransfer] [rskTx:%s] error during validationsForRegisterBtcTransaction", ByteUtil.toHexString(btcTxHash.getBytes()));
+            logger.warn(errorMesage);
+            throw new RegisterBtcTransferRegisterBtcTransValidationException(errorMesage);
+        }
+
+        if (!provider.isFastBridgeFederationDerivationHashUsed(derivationArgumentsHash)) {
+            logger.warn("[registerBtcTransfer] [btcTxHash:{}] derivationArgumentsHash is already saved in BridgeStorageProvider ", btcTxHash);
```

Comment:
Condition here should be true, we are checking if derivation arguments hash was already used